engine/src/test/java/com/datatorrent/stram/engine/GenericNodeTest.java

```diff
@@ -38,10 +49,40 @@
  */
 public class GenericNodeTest
 {
-  public static class GenericOperator implements Operator
+  @Rule
+  public FSTestWatcher testMeta = new FSTestWatcher();
+
+  public static class FSTestWatcher extends TestWatcher
+  {
+    public org.junit.runner.Description desc;
+
+    public String getDir()
+    {
```

Comment:
This method gets called twice for a test. Instead have a directory variable defined and initialize that variable with "target/" + desc.getClassName() + "/" + desc.getMethodName() in starting(). You don't need to store description variable